File: C++/src/Ecuacion.cpp
#include "Ecuacion.h"s
#include <math.h>

Ecuacion::Ecuacion()
{
    // al inicializar el constructor se definiran los valores iniciales de los atributos
    a = 0;
    b = 0;
    c = 0;
}

Ecuacion::~Ecuacion()
{
    //dtor
}

// metodos Setters y Getters de los atributos de la clase
void Ecuacion::setA(float val)
{
    //IGUALAMOS EL PARAMETRO A LA VARIABLE (A)
    a = val;
}
float Ecuacion::getA()
{
    return a;
}
void Ecuacion::setB(float val)
{
    b = val;
}
float Ecuacion::getB()
{
    return b;
}
void Ecuacion::setC(float val)
{
    c = val;
}
float Ecuacion::getC()
{
    return c;
}

// solucion 1
float Ecuacion::resultadoUno()
{
    float res=0;
    res = (-getB() + (sqrt(pow(getB(),2)-(4 *getA()*getC()) ))) / (2*getA());
    return res;
}

// solucion 2
float Ecuacion::resultadoDos()
{
    float res=0;
    res = (-getB() - (sqrt(pow(getB(),2)-(4 * getA() * getC())))) / (2*getA());
    return res;
}

File: Java/src/Ecuacion.java
public class Ecuacion
{
    // se declaran los atributos privados
    private float a;
    private float b;
    private float c;
    
    public Ecuacion()
    {
        // al inicializar el constructor se definiran los valores iniciales de los atributos
        a = 0;
        b = 0;
        c = 0;
    }
    
    // metodos Setters y Getters de los atributos de la clase
    public void setA(float val)
    {
        a = val;
    }
    public float getA()
    {
        return a;
    }
    
    public void setB(float val)
    {
        b = val;
    }
    public float getB()
    {
        return b;
    }
    public void setC(float val)
    {
        c = val;
    }
    public float getC()
    {
        return c;
    }
    
    // solución 1
    public double resultadoUno()
    {
        // declaramos la variable que devuelve el resultado
        double solu;
        
        // ecuacion de segundo grado
        solu= (-b + (Math.sqrt((b*b)-(4*a*c))))  / (2*a);
        
        // retorna el resultado
        return solu;
                
    }
    
    // solución 2
    public double resultadoDos()
    {
        // declaramos la variable que devuelve el resultado
        double sol;
        
        // ecuacion de segundo grado
        sol= (-b - (Math.sqrt((b*b)-(4*a*c))) ) / (2*a);
        
        // retorna el resultado
        return sol;
    }
}